### DISCUSSION TITLE
Make FileManager parameters available in RichTextEditor

### DISCUSSION THREAD

**Discussion by mdmontesinos** [oqtane/oqtane.framework#2383] [Ideas]
It would be nice that you are able to customize the FileManager used by the RichTextEditor, and consequently, the HtmlText.

For example, I want to have an HtmlText module whose FileManager can only access a specific folder. Therefore, I would use a file manager like this:
```c#
<FileManager @ref="_fileManager" Filter="@Constants.ImageFiles" FolderId="_folderId" ShowFolders="false"></FileManager>
```
However, as the FileManager reference is private in the RichTextEditor, I can't pass any parameters to it. It would be unnecessary to create a custom component (basically a copy of both RichTextEditor and HtmlText) just to add this feature.

Perhaps a good solution is to make the FileManager reference in RichText a parameter itself, so it can be configured beforehand, and sent to the component.

**Comment by sbwalker** [comment]
Additional parameters could be added to the RichTextEditor which would then be passed to the FileManager. The Html/Text module Settings could be enhanced to allow a user to specify those parameters through the UI.

**Reply by mdmontesinos** [reply]
Is that the preferred approach? I mean generating additional parameters in the RichTextEditor instead of exposing the internal FileManager.

In any case, I'll take a look at it and try to submit a PR when ready.

**Reply by sbwalker** [reply]
Allowing FileManager to be passed in as a parameter is probably better as it allows more flexibility in the future.
